Source/missiles.cpp

```diff
@@ -3225,56 +3287,59 @@ void MI_Acidpud(int i)
 void MI_Firewall(int i)
 {
 	constexpr int ExpLight[14] = { 2, 3, 4, 5, 5, 6, 7, 8, 9, 10, 11, 12, 12 };
+	auto &missile = Missiles[i];
 
-	Missiles[i]._mirange--;
-	if (Missiles[i]._mirange == Missiles[i]._miVar1) {
-		SetMissDir(i, 1);
-		Missiles[i]._miAnimFrame = GenerateRnd(11) + 1;
+	missile._mirange--;
+	if (missile._mirange == missile._miVar1) {
+		SetMissDir(missile, 1);
+		missile._miAnimFrame = GenerateRnd(11) + 1;
 	}
-	if (Missiles[i]._mirange == Missiles[i]._miAnimLen - 1) {
-		SetMissDir(i, 0);
-		Missiles[i]._miAnimFrame = 13;
-		Missiles[i]._miAnimAdd = -1;
+	if (missile._mirange == missile._miAnimLen - 1) {
+		SetMissDir(missile, 0);
+		missile._miAnimFrame = 13;
+		missile._miAnimAdd = -1;
 	}
-	CheckMissileCol(i, Missiles[i]._midam, Missiles[i]._midam, true, Missiles[i].position.tile, true);
-	if (Missiles[i]._mirange == 0) {
-		Missiles[i]._miDelFlag = true;
-		AddUnLight(Missiles[i]._mlid);
+	CheckMissileCol(i, missile._midam, missile._midam, true, missile.position.tile, true);
+	if (missile._mirange == 0) {
+		missile._miDelFlag = true;
+		AddUnLight(missile._mlid);
 	}
-	if (Missiles[i]._mimfnum != 0 && Missiles[i]._mirange != 0 && Missiles[i]._miAnimAdd != -1 && Missiles[i]._miVar2 < 12) {
-		if (Missiles[i]._miVar2 == 0)
-			Missiles[i]._mlid = AddLight(Missiles[i].position.tile, ExpLight[0]);
-		ChangeLight(Missiles[i]._mlid, Missiles[i].position.tile, ExpLight[Missiles[i]._miVar2]);
-		Missiles[i]._miVar2++;
+	if (missile._mimfnum != 0 && missile._mirange != 0 && missile._miAnimAdd != -1 && missile._miVar2 < 12) {
+		if (missile._miVar2 == 0)
+			missile._mlid = AddLight(missile.position.tile, ExpLight[0]);
+		ChangeLight(missile._mlid, missile.position.tile, ExpLight[missile._miVar2]);
+		missile._miVar2++;
 	}
 	PutMissile(i);
 }
 
 void MI_Fireball(int i)
 {
-	FireballUpdate(i, Missiles[i].position.velocity, false);
+	auto &missile = Missiles[i];
+	FireballUpdate(i, missile.position.velocity, false);

```

Comment:
It would be good to have FireballUpdate take `MissileStruct&` as the first argument to make this change worthwhile. The current use with a call to PutMissile at the end of that function makes it difficult to do this refactoring now though so happy for that to come later.

Reply:
Yeah, I think this should be deferred til round 4 or 5 :)